lib/core/metadata.hpp

```diff
@@ -0,0 +1,147 @@
+/**
+ * @file
+ * @brief The kotekan buffer metadata containers and pools.
+ * Most of these functions are used by buffer.c internally and not
+ * intended for use outside of the buffer content.
+ * - metadataContainer
+ * -- create_metadata
+ * -- delete_metadata
+ * -- reset_metadata_object
+ * -- increment_metadata_ref_count
+ * -- decrement_metadata_ref_count
+ * -- lock_metadata
+ * -- unlock_metadata
+ * - metadataPool
+ * -- create_metadata_pool
+ * -- delete_metadata_pool
+ * -- request_metadata_object
+ * -- return_metadata_to_pool
+ */
+
+#ifndef KOTEKAN_METADATA_HPP
+#define KOTEKAN_METADATA_HPP
+
+#include "factory.hpp"
+#include "kotekanLogging.hpp"
+
+#include "json.hpp" // for json
+
+#include <memory>
+#include <mutex>
+#include <stdint.h> // for uint32_t
```

Comment:
Maybe `cstdint` and `cstdio`? Or `iostream` instead?

Reply:
Actually, `size_t` should also be in `cstdint` or `cstddef`.

Reply:
fixed!